### PR TITLE
MM-35244: updated telemetry

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
I combed through the actions and properties of our current telemetry and the in-product experience, adding the following:
* remove timeline event
* ChecklistItem properties
* updated Incident and Playbook properties
* timeline_tab_filter_selected event to track additional timeline interactions

Proposing including this in v1.9 -- cc @prapti.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-35244

### CODE REVIEW

File: server/incident/service.go [hunk 1]
```diff
@@ -834,6 +839,8 @@ func (s *ServiceImpl) ModifyCheckedState(incidentID, userID, newState string, ch
 		return errors.Wrapf(err, "failed to update incident, is now in inconsistent state")
 	}
 
+	s.telemetry.ModifyCheckedState(incidentID, userID, itemToCheck, incidentToModify.CommanderUserID == userID)
```
Comment (lieut-data) [inline server/incident/service.go:842]: Moved this out from underneath a comment that wasn't relevant, until after the change is actually effected.

File: server/incident/service.go [hunk 2]
```diff
@@ -930,6 +937,8 @@ func (s *ServiceImpl) SetAssignee(incidentID, userID, assigneeID string, checkli
 		return errors.Wrapf(err, "failed to update incident; it is now in an inconsistent state")
 	}
 
+	s.telemetry.SetAssignee(incidentID, userID, itemToCheck)
```
Comment (lieut-data) [inline server/incident/service.go:940]: Move this up before the timeline events (that might fail).

File: server/incident/service.go [hunk 3]
```diff
@@ -993,6 +1000,8 @@ func (s *ServiceImpl) RunChecklistItemSlashCommand(incidentID, userID string, ch
 		return "", errors.Wrapf(err, "failed to update incident recording run of slash command")
 	}
 
+	s.telemetry.RunTaskSlashCommand(incidentID, userID, itemToRun)
```
Comment (lieut-data) [inline server/incident/service.go:1003]: Again, up before timeline events.

File: server/telemetry/rudder.go [hunk 4]
```diff
@@ -107,14 +109,22 @@ func incidentProperties(incdnt *incident.Incident, userID string) map[string]int
 	return map[string]interface{}{
```
Comment (lieut-data) [inline server/telemetry/rudder.go:109]: I know we have plans to annotate the structs here, but I didn't want to add too much complexity to this first pass.

File: server/telemetry/rudder.go [hunk 5]
```diff
@@ -247,15 +270,27 @@ func playbookProperties(pbook playbook.Playbook, userID string) map[string]inter
...
 		"UsesReminderMessageTemplate": pbook.ReminderMessageTemplate != "",
+		"ReminderTimerDefaultSeconds": pbook.ReminderTimerDefaultSeconds,
+		"NumInvitedUserIDs":           len(pbook.InvitedUserIDs),
+		"NumInvitedGroupIDs":          len(pbook.InvitedGroupIDs),
+		"InviteUsersEnabled":          pbook.InviteUsersEnabled,
+		"DefaultCommanderID":          pbook.DefaultCommanderID,
+		"DefaultCommanderEnabled":     pbook.DefaultCommanderEnabled,
+		"AnnouncementChannelID":       pbook.AnnouncementChannelID,
+		"AnnouncementChannelEnabled":  pbook.AnnouncementChannelEnabled,
+		"NumWebhookOnCreationURLs":    len(strings.Split(pbook.WebhookOnCreationURL, ",")),
```
Comment (lieut-data) [inline server/telemetry/rudder.go:292]: I'm assuming we'll add support for multiple webhooks, so splitting on commas.

**Review by crspeller — APPROVED** [review]

File: server/telemetry/rudder.go (hunk 5 above)
Comment (cpoile) [reply server/telemetry/rudder.go:292]: just wondering if commas are allowed in urls -- quick search says technically they are (in filenames?). Don't know if we want to use a space instead?

**Review by cpoile — APPROVED** [review]
One non-blocking question, nice!

File: server/telemetry/rudder.go (hunk 5 above)
Comment (lieut-data) [reply server/telemetry/rudder.go:292]: Ah, good catch -- I think the server /actually/ uses newlines, so I'll split on that instead.

**Pull request merged**